### PR TITLE
Add logfactorial.c

### PR DESCRIPTION
**Pull request by ritwik12**
Add implementation for log_of_factorial in C.

### CODE REVIEW

File: code/mathematical-algorithms/basic/log_of_factorial/logfactorial.c [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#include <math.h>
+#include <stdio.h>
+
+double log_factorial(int n){
```
Comment (arnavb) [inline code/mathematical-algorithms/basic/log_of_factorial/logfactorial.c]: Shouldn't the return value be an int, since this is factorial?

**Comment by ritwik12** [comment]
@arnavb Thanks for the review.
Please correct logfactorial.cpp, as I did my mistake from there.
For a value of n=4 we should get an answer 2.
But in .cpp we are getting 3.17 something

**Comment by arnavb** [comment]
For n = 4, you should get 24, right?

**Comment by ritwik12** [comment]
@arnavb  it is log of factorial.
You are confusing it for factorial

**Comment by ritwik12** [comment]
So please tell me the output for n=4

File: code/mathematical-algorithms/basic/log_of_factorial/logfactorial.c [hunk 2]
```diff
@@ -1,8 +1,8 @@
 #include <math.h>
```
Comment (yatharthshahjpr) [inline code/mathematical-algorithms/basic/log_of_factorial/logfactorial.c]: Please add comment  /* Part of Cosmos by OpenGenus Foundation */ at the top of every code you contribute to this project  ! 👍

**Comment by ritwik12** [comment]
According to me log(4!)= log(1)+log(2)+log(3)+log(4)

File: code/mathematical-algorithms/basic/log_of_factorial/logfactorial.c (hunk 2 above)
Comment (ritwik12) [reply code/mathematical-algorithms/basic/log_of_factorial/logfactorial.c]: Alright, I will.

**Comment by arnavb** [comment]
Oh, then my review doesn't apply. Disregard it completely.

**Comment by ritwik12** [comment]
I got my doubt cleared, It is ln(n!), I thought lg(n!)

**Comment by ritwik12** [comment]
It's good to go from my side :)

**Comment by ritwik12** [comment]
@AdiChat  Also please tell me how could I join OpenGenus?
This is my 3-4th contribution.

**Comment by arnavb** [comment]
@ritwik12 AdiChat will send you an invite to the organization.

**Comment by ritwik12** [comment]
@arnavb Ok, Thanks.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**